bin/node-template/node/src/service.rs

```diff
@@ -24,100 +20,97 @@ native_executor_instance!(
 
-		let builder = sc_service::ServiceBuilder::new_full::<
-			node_template_runtime::opaque::Block,
-			node_template_runtime::RuntimeApi,
-			crate::service::Executor
-		>($config)?
-			.with_select_chain(|_config, backend| {
-				Ok(sc_consensus::LongestChain::new(backend.clone()))
-			})?
-			.with_transaction_pool(|builder| {
-				let pool_api = sc_transaction_pool::FullChainApi::new(
-					builder.client().clone(),
-				);
-				Ok(sc_transaction_pool::BasicPool::new(
-					builder.config().transaction_pool.clone(),
-					std::sync::Arc::new(pool_api),
-					builder.prometheus_registry(),
-				))
-			})?
-			.with_import_queue(|
-				_config,
-				client,
-				mut select_chain,
-				_transaction_pool,
-				spawn_task_handle,
-				registry,
-			| {
-				let select_chain = select_chain.take()
-					.ok_or_else(|| sc_service::Error::SelectChainRequired)?;
+		let (client, backend, keystore, task_manager) = sc_service::new_full_parts::<Block, RuntimeApi, Executor>(&$config)?;
+		let client = Arc::new(client);
 
-				let (grandpa_block_import, grandpa_link) = sc_finality_grandpa::block_import(
-					client.clone(),
-					&(client.clone() as Arc<_>),
-					select_chain,
-				)?;
+		let select_chain = sc_consensus::LongestChain::new(backend.clone());
 
-				let aura_block_import = sc_consensus_aura::AuraBlockImport::<_, _, _, AuraPair>::new(
-					grandpa_block_import.clone(), client.clone(),
-				);
+		let pool_api = sc_transaction_pool::FullChainApi::new(client.clone());
+		let transaction_pool = sc_transaction_pool::BasicPool::new_full(
+			$config.transaction_pool.clone(),
+			std::sync::Arc::new(pool_api),
+			$config.prometheus_registry(),
+			task_manager.spawn_handle(),
+			client.clone(),
+		);
 
-				let import_queue = sc_consensus_aura::import_queue::<_, _, _, AuraPair, _>(
-					sc_consensus_aura::slot_duration(&*client)?,
-					aura_block_import,
-					Some(Box::new(grandpa_block_import.clone())),
-					None,
-					client,
-					inherent_data_providers.clone(),
-					spawn_task_handle,
-					registry,
-				)?;
+		let (grandpa_block_import, grandpa_link) = sc_finality_grandpa::block_import(
+			client.clone(), select_chain.clone(),
+		)?;
 
-				import_setup = Some((grandpa_block_import, grandpa_link));
+		let aura_block_import = sc_consensus_aura::AuraBlockImport::<_, _, _, AuraPair>::new(
+			grandpa_block_import.clone(), client.clone(),
+		);
 
-				Ok(import_queue)
-			})?;
+		let import_queue = sc_consensus_aura::import_queue::<_, _, _, AuraPair, _>(
+			sc_consensus_aura::slot_duration(&*client)?,
+			aura_block_import,
+			Some(Box::new(grandpa_block_import.clone())),
+			None,
+			client.clone(),
+			inherent_data_providers.clone(),
+			&task_manager.spawn_handle(),
+			$config.prometheus_registry(),
+		)?;
 
-		(builder, import_setup, inherent_data_providers)
+		(
+			client, backend, keystore, task_manager, inherent_data_providers, select_chain,
+			transaction_pool, grandpa_block_import, grandpa_link, import_queue
+		)
```

Comment:
The return type is going to be complicated. Probably we should find a way to get a single type... idk... like `BaseServiceComponents` maybe?

Reply:
No, the point of this refactor should be that we not again introduce these types that hold opinionated fields.